platforms/platform-core/src/CoreRequest.ts

```diff
@@ -18,8 +25,12 @@ export class CoreRequest extends JovoRequest {
     return this.locale;
   }
 
+  getInput(): JovoInput {
+    return new JovoInputBuilder(super.getInput()).set('nlu', this.input?.nlu).build();
+  }
+
   getIntent(): JovoInput['intent'] {
-    return this.input?.intent;
+    return this.input?.nlu?.intent || this.input?.intent;
```

Comment:
The root `intent` should be prioritized:

```
return this.input?.intent || this.input?.nlu?.intent;
```

Reply:
Done